skyportal/facility_apis/lt.py

```diff
@@ -0,0 +1,325 @@
+        airmass_const = etree.Element('AirmassConstraint', maximum=str(LT_SETTINGS['MAX_AIRMASS']))
+
+        sky_const = etree.Element('SkyConstraint')
+        etree.SubElement(sky_const, 'Flux').text = str(LT_SETTINGS['MAX_SKYBRI'])
+        etree.SubElement(sky_const, 'Units').text = 'magnitudes/square-arcsecond'
+
+        seeing_const = etree.Element('SeeingConstraint',
+                                     maximum=(str(LT_SETTINGS['MAX_SEEING'])),
+                                     units='arcseconds')
+
+        photom_const = etree.Element('ExtinctionConstraint')
+        etree.SubElement(photom_const, 'Clouds').text = LT_SETTINGS['PHOTOMETRIC']
+
+        date_const = etree.Element('DateTimeConstraint', type='include')
+        start = request.payload["start_date"] + 'T00:00:00+00:00'
+        end = request.payload["end_date"] + 'T00:00:00+00:00'
+        etree.SubElement(date_const, 'DateTimeStart', system='UT', value=start)
+        etree.SubElement(date_const, 'DateTimeEnd', system='UT', value=end)
+
+        return [airmass_const, sky_const, seeing_const, photom_const, date_const]
+
+    def _build_target(self, request):
+        target = etree.Element('Target', name=request.obj.id)
+        c = SkyCoord(ra=request.obj.ra*u.degree,
+                     dec=request.obj.dec*u.degree)
+        coordinates = etree.SubElement(target, 'Coordinates')
+        ra = etree.SubElement(coordinates, 'RightAscension')
+        etree.SubElement(ra, 'Hours').text = str(int(c.ra.hms.h))
+        etree.SubElement(ra, 'Minutes').text = str(int(c.ra.hms.m))
+        etree.SubElement(ra, 'Seconds').text = str(c.ra.hms.s)
+
+        dec = etree.SubElement(coordinates, 'Declination')
+        sign = '+' if c.dec.signed_dms.sign == 1.0 else '-'
+        etree.SubElement(dec, 'Degrees').text = sign + str(int(c.dec.signed_dms.d))
+        etree.SubElement(dec, 'Arcminutes').text = str(int(c.dec.signed_dms.m))
+        etree.SubElement(dec, 'Arcseconds').text = str(c.dec.signed_dms.s)
+        etree.SubElement(coordinates, 'Equinox').text = 'J2000'
+        return target
+
+    def _build_inst_schedule(self, payload, request):
+
+        if request.payload["instrument_type"] == "IOO":
+            exposure_type = request.payload["exposure_type"]
+            exposure_type_split = exposure_type.split("x")
+            exp_count = int(exposure_type_split[0])
+            exp_time = int(exposure_type_split[1][:-1])
+            for filt in request.payload["observation_type"]:
+                payload.append(self._build_IOO_schedule(request, 
+                                                        filt,
+                                                        exp_time,
+                                                        exp_count))
+        elif request.payload["instrument_type"] == "SPRAT":
+            self._build_SPRAT_schedule(payload, request)
+
+    def _build_IOO_schedule(self, request, filt, exp_time, exp_count):
```

Comment:
We should break these out into subclasses. I'd recommend a base `LTAPI`, and then each instrument gets its own subclass, e.g., `IOOAPI`, `SPRATAPI`, &c